src/Core/Service/Api/PermissionController.php

```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Service\Api;
+
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Service\PermissionsService;
+use Shopware\Core\Service\ServiceRegistryClient;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Attribute\Route;
+
+/**
+ * @internal only for use by the service-system
+ */
+#[Route(defaults: ['_routeScope' => ['api']])]
+#[Package('framework')]
+readonly class PermissionController
+{
+    public function __construct(
+        private ServiceRegistryClient $serviceRegistryClient,
+        private PermissionsService $permissionsService,
+    ) {
+    }
+
+    #[Route(path: '/api/services/permissions', name: 'api.services.permissions', defaults: ['auth_required' => true, '_acl' => ['system.system_config']], methods: ['GET'])]
+    public function getPermissionsRevisions(): JsonResponse
```

Comment:
do we need this endpoint? `permissions.json` - would be publicly available. I thought the FE module could just call that directly

Reply:
Would certainly be feasible but opens up for other potential sources of errors like e.g. CORS that we can circumvent with doing server-to-server communication.

Plus we can map to our very own `errorCode` via the `ShopwareHttpException` which we can handle in a streamlined way in the FE.

I'm leaning towards keeping this endpoint 😄

Reply:
Talked to @SebastianFranze and we will fetch it directly from the registry shopware/service-registry#285 and **not proxy** through Shopware Admin API.